### PR TITLE
Used a conditional type for Sinon's stub class typings

### PR DESCRIPTION
**Pull request by JoshuaKGoldberg**
I wonder - is it soon enough to require TS 2.8 to use Sinon? It's also necessary for all Sinon-dependent types to be on 2.8 too, right?

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/9f4c75126167d0d8af759f58405d53d983e94ad0/types/sinon/index.d.ts#L570
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by JoshuaKGoldberg** [comment]
Err. https://travis-ci.org/DefinitelyTyped/DefinitelyTyped/builds/355711711?utm_source=github_status&utm_medium=notification

> sinon-chrome depends on chrome but has a lower required TypeScript version

`sinon-chrome` has 2.8; `chrome` has 2.4. I'm guessing TS 2.8 types aren't usable here yet? @RyanCavanaugh

**Comment by nicojs** [comment]
@JoshuaKGoldberg awesome improvement! I've thought of this myself, but I think we should wait for typescript 2.8 to officially release.

:+1:

**Comment by stevehipwell** [comment]
@JoshuaKGoldberg - This looks great but I'd also suggest that we wait for the TypeScript 2.8 release.

**Comment by RyanCavanaugh** [comment]
You won't be able to use 2.8 features on DT until TS2.8 is actually out.

Once TS2.8 is out, this will ship under the `@TS28` tag but won't be under `@latest` until 6 weeks later (@andy-ms please confirm I got that right?)

**Comment by andy-ms** [comment]
@RyanCavanaugh Basically right. We don't have a builtin timer mechanism but `definitelytyped-header-parser` has an [`isPrerelease`](https://github.com/Microsoft/definitelytyped-header-parser/blob/master/index.ts#L21) function we can edit, and our policy has generally been to leave a version as prerelease until at least a month after release.

**Comment by RyanCavanaugh** [comment]
@JoshuaKGoldberg the bot is going to close this but please reopen in a few weeks. Thanks!

File: types/puppeteer/index.d.ts [hunk 1]
```diff
@@ -11,6 +11,18 @@
 import { EventEmitter } from "events";
 import { ChildProcess } from "child_process";
 
+// Stub DOM types when the dom lib isn't included
```
Comment (JoshuaKGoldberg) [inline types/puppeteer/index.d.ts]: Unrelated changes, whoops.

Comment (andy-ms) [reply types/puppeteer/index.d.ts]: If it's not needed could you make another PR removing it?

Comment (JoshuaKGoldberg) [reply types/puppeteer/index.d.ts]: I'll update this PR in a bit
